### PR TITLE
Add Volunteer Model

### LINKED ISSUE #47: As a single administrator of a small NPOs, I can add volunteer to the list of volunteers

### LINKED ISSUE #48: As a single administrator of a small NPOs, I can delete volunteers

### PR DESCRIPTION
**Pull request by zhacatomn**
Fixes #47, fixes #48 
 
## Added
- New `Volunteer` model
- New `AddVolunteerCommand` and `AddVolunterCommandParser` models
- New tests for `Volunteer` model, including a  `VolunteerBuilder`. (I also added an abstract class called `PersonBuilderScaffold`. This is to allow better inheritance using the builder pattern.)
- Modified name of `PREFIX_ELDERLY_NRIC` and `PREFIX_VOLUNTEER_NRIC` to `PREFIX_NRIC_ELDERLY` and `PREFIX_NRIC_VOLUNTEER`
- Refactored out `ModelStub` to `CommandTestUtil` to minimise code duplication

## Left To Do
- [x] Tests for `AddVolunteerCommand`

## Things to note
- We should probably just add NRIC and age to the `Person` model: right now there's quite a bit of code duplication between the `Volunteer` and `Elderly` classes. Will do in a separate PR probably
- The v1.2 User Guide seems to not be fully in-line with the current implementation: the `add_volunteer` command in the guide is missing `p/`, `t/` and `ag/` (Phone number, tag, and age respectively)
- `delete` command currently is one command for both the elderly and volunteers using the list index. Should break this up into 2 commands using NRIC instead as per the guide. To do in a future PR.
- Still unsure whether we are adding "Medical training" as a tag, or as an attribute to the `Volunteer` class

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddVolunteerCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        return new AddVolunteerCommand(volunteer);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```
Comment (Zxun2) [inline src/main/java/seedu/address/logic/parser/AddVolunteerCommandParser.java:69]: There's an existing method that does the same thing in `commons/util`. Can you update this file to use that instead? thanks 👍

File: src/main/java/seedu/address/model/person/Volunteer.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import seedu.address.model.person.information.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents an Volunteer in the database.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Volunteer extends Person {
+    private final Nric nric;
+    private final Age age;
```
Comment (Zxun2) [inline src/main/java/seedu/address/model/person/Volunteer.java:19]: If both `Volunteer` and `Elderly` have the `Nric` and `Age`, we should make them part of `Person`. What do you think?

**Review by Zxun2 — APPROVED** [review]
Great work! We can merge this in once the changes are made

**Comment by Zxun2** [comment]
> We should probably just add NRIC and age to the Person model: right now there's quite a bit of code duplication between the Volunteer and Elderly classes. Will do in a separate PR probably

Yup, I agree. I vaguely remember that `Volunteer` should have some attributes, but I can't recall now lol. 

> Still unsure whether we are adding "Medical training" as a tag, or as an attribute to the Volunteer class

Btw, @wz2k can you confirm that the `MedicationQualificationTag` is now no longer a tag? If so, can you create an issue and update the current one in the main branch?

File: src/main/java/seedu/address/logic/commands/AddPairCommand.java [hunk 3]
```diff
@@ -17,11 +17,11 @@ public class AddPairCommand extends Command {
...
             + "Parameters: "
-            + PREFIX_ELDERLY_NRIC + "ELDERLY ID "
-            + PREFIX_VOLUNTEER_NRIC + "VOLUNTEER ID "
+            + PREFIX_NRIC_ELDERLY + "ELDERLY ID "
+            + PREFIX_NRIC_VOLUNTEER + "VOLUNTEER ID "
             + "Example: " + COMMAND_WORD + " "
-            + PREFIX_ELDERLY_NRIC + "1234 "
-            + PREFIX_VOLUNTEER_NRIC + "5678 ";
+            + PREFIX_NRIC_ELDERLY + "1234 "
+            + PREFIX_NRIC_VOLUNTEER + "5678 ";
```
Comment (heeeyi) [inline src/main/java/seedu/address/logic/commands/AddPairCommand.java:25]: Am I understanding correctly that we need to use elderly and volunteer's NRIC for pairing, so we need to separate their prefixes?

**Review by heeeyi — COMMENTED** [review]
I feel everything is fine, that's really nice job!

File: src/main/java/seedu/address/logic/parser/AddVolunteerCommandParser.java (hunk 1 above)
Comment (zhacatomn) [reply src/main/java/seedu/address/logic/parser/AddVolunteerCommandParser.java]: Oh nice, I'll do that.

File: src/main/java/seedu/address/logic/commands/AddPairCommand.java (hunk 3 above)
Comment (zhacatomn) [reply src/main/java/seedu/address/logic/commands/AddPairCommand.java:25]: Yup I believe that was the idea. We had this in the user guide as well, so I decided to follow it.

**Pull request merged**